packages/oauth/oauth-client/src/oauth-server-factory.ts

```diff
@@ -17,19 +21,47 @@ export class OAuthServerFactory {
     readonly dpopNonceCache: DpopNonceCache,
   ) {}
 
-  async fromIssuer(issuer: string, dpopKey: Key, options?: GetCachedOptions) {
+  /**
+   * @param authMethod `undefined` means that we are restoring a session that
+   * was created before we started storing the `authMethod` in the session. In
+   * that case, we will use the first key from the keyset.
+   *
+   * Support for this might be removed in the future.
+   *
+   * @throws see {@link OAuthServerFactory.fromMetadata}
+   */
+  async fromIssuer(
+    issuer: string,
+    authMethod: 'legacy' | ClientAuthMethod,
+    dpopKey: Key,
+    options?: GetCachedOptions,
+  ) {
     const serverMetadata = await this.resolver.getAuthorizationServerMetadata(
       issuer,
       options,
     )
-    return this.fromMetadata(serverMetadata, dpopKey)
+
+    if (authMethod === 'legacy') {
```

Comment:
Because we were previously not storing the `authMethod` in the session data, we provide a backwards compatible implementation by computing it here.